code/modules/clothing/spacesuits/hardsuit.dm

```diff
@@ -365,6 +365,8 @@
 	armor_type = /datum/armor/engine_atmos
 	heat_protection = HEAD												//Uncomment to enable firesuit protection
 	max_heat_protection_temperature = FIRE_IMMUNITY_MAX_TEMP_PROTECT
+	allowed = list(/obj/item/flashlight, /obj/item/tank/internals, /obj/item/t_scanner,
```

Comment:
This is a helmet you're trying to add storage to. Remove this as well, it does nothing at best and causes unintended behavior at worst. 

If you want to add this to the hardsuit as I believe your intentions were, that would be fine, but that will be around line 390 or so.